src/conf.c

```diff
@@ -271,20 +271,18 @@ ConfSetFinal(const char *name, char *val)
 }
 
 /**
- * \brief Retrieve the value of a configuration node.
+ * \brief Get the value of a configuration node.
  *
- * This function will return the value for a configuration node based
- * on the full name of the node.  It is possible that the value
- * returned could be NULL, this could happen if the requested node
- * does exist but is not a node that contains a value, but contains
- * children ConfNodes instead.
+ * Note on failure: This function will return for failure if the named
+ * configuration node does not exist, or if it does exist and its
+ * value is 0.
  *
  * \param name Name of configuration parameter to get.
  * \param vptr Pointer that will be set to the configuration value parameter.
  *   Note that this is just a reference to the actual value, not a copy.
  *
- * \retval 1 will be returned if the name is found, otherwise 0 will
- *   be returned.
+ * \retval 1 will be returned if the name is found and the value is not NULL,
```

Comment:
How would one retrieve a node that contains a list now? There we won't have a value, so getting the node will fail. This is visible in src/suricata.c:2079 where the check now leads to failure on a valid config